folly/Portability.h

```diff
@@ -284,6 +284,66 @@ inline size_t malloc_usable_size(void* ptr) {
 # define FOLLY_HAS_RTTI 1
 #endif
 
+#ifdef _MSC_VER
+#include <pthread.h>
+// We implement a sane comparison operand for
+// pthread_t and an integer so that it may be
+// compared agains 0.
+
+inline bool operator ==(pthread_t ptA, unsigned int b) {
+  if (ptA.p == NULL) {
+    return b == 0;
+  }
+  return pthread_getw32threadid_np(ptA) == b;
+}
+
+inline bool operator !=(pthread_t ptA, unsigned int b) {
+  if (ptA.p == NULL) {
+    return b != 0;
+  }
+  return pthread_getw32threadid_np(ptA) != b;
+}
+
+inline bool operator ==(pthread_t ptA, pthread_t ptB) {
+  return pthread_equal(ptA, ptB) != 0;
+}
+
+inline bool operator !=(pthread_t ptA, pthread_t ptB) {
+  return pthread_equal(ptA, ptB) == 0;
+}
+
+inline bool operator !(pthread_t ptA) {
+  return ptA == 0;
+}
+
+#define pthread_t_init (pthread_t{NULL, 0})
```

Comment:
I'll go with pthread_zero then. I believe I have an additional bit locally for this, and I need to implement a hasher for it for std::hash, as HHVM uses pthread_t as a key.